firebolt_provider/hooks/firebolt.py

```diff
@@ -116,40 +105,36 @@ def _get_conn_params(self) -> Dict[str, Optional[str]]:
         """
         conn_id = getattr(self, self.conn_name_attr)
         conn = self.get_connection(conn_id)
-        database = conn.schema
+        database = self.database or conn.schema
 
-        engine_name = self.engine_name or conn.extra_dejson.get("engine_name", None)
+        engine_name = self.engine_name or conn.host
+        api_endpoint = conn.extra_dejson.get("api_endpoint", DEFAULT_API_URL)
+        account_name = conn.extra_dejson.get("account_name", None)
+        if not account_name:
+            raise FireboltError("Account name is missing")
 
-        engine_url = None
-        if engine_name and "." in engine_name:
-            engine_name, engine_url = None, engine_name
+        if not (conn.login and conn.password):
+            raise FireboltError("Either cliend id or client secret is missing")
 
-        api_endpoint = conn.extra_dejson.get("api_endpoint", None)
-        account_name = conn.extra_dejson.get("account_name", None)
         conn_config = {
-            "username": conn.login,
-            "password": conn.password or "",
-            "api_endpoint": api_endpoint or DEFAULT_API_URL,
-            "database": self.database or database,
+            "client_id": conn.login,
```

Comment:
I'd define variables `client_id` and `client_secret` next to the `account_name` and use them later. This will separate the initialisation section from the logic.

Reply:
I think that's a bit of overhead to just create a new variable for the same value. WDYT?